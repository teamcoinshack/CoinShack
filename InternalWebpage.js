import React from 'react';
import { WebView} from 'react-native'

export default class InternalWebpage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      url: '',
    }
  }

  componentDidMount() {
    const { navigation } = this.props;
    const url = navigation.getParam('url', null);
    this.setState({
      url: url,
    });
  }

  render() {
    return (
      <WebView
        source={{ uri: this.state.url }}
        style={{ marginTop: 20 }}
      />
    );
  }
}
