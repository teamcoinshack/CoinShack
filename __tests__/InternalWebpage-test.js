/**
 * @format
 */

import 'react-native';
import React from 'react';
import InternalWebpage from '../screens/InternalWebpage.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
const paramFunction = (param) => {
  if (param === 'url') {
    return 'www.google.com';
  } else {
    return null;
  }
}

const createTestProps = (props: Object) => ({
  navigation: {
    getParam: paramFunction
  },
  ...props
});

jest.mock('react-native-google-signin', () => {});
test('renders correctly', () => {
  let props: any;
  props = createTestProps({});
  const tree = renderer.create(<InternalWebpage {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
