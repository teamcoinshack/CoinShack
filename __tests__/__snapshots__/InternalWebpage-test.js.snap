// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <RNCWKWebView
    cacheEnabled={true}
    messagingEnabled={false}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    source={
      Object {
        "uri": "www.google.com",
      }
    }
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        Object {
          "marginTop": 0,
        },
      ]
    }
    useSharedProcessPool={true}
  />
</View>
`;
